util/src/result.rs

```diff
@@ -2,6 +2,11 @@
 #[macro_export]
 macro_rules! attempt {
     ($block:expr) => {
-        (|| -> ::anyhow::Result<_> { $block })()
+        (|| -> Result<_, ::rosenpass_util::SodiumError> {
```

Comment:
This is obviously not a good idea. Either infer the error type or get rid of the attempt macro

Reply:
infered the error type